frontend/app/src/scenes/Home/scenes/UpsertCategory/index.tsx

```diff
@@ -50,7 +57,20 @@ class UpsertCategory extends React.Component<
       )
       .then((result: AxiosResponse<Category>) => {
         if (result.status == 200) {
-          this.setState({ isLoading: false, category: result.data });
+          this.setState({
+            isLoading: false,
+            category: result.data,
+          });
+          // Gets the translations for the fetched category and setting them to the state using a map
+          this.state.category?.translations.forEach((translation) => {
```

Comment:
is there any specific reason to iterate the state? I think at this stage `state.category?.translations` is an empty map. Iterating it doesn't do anything.

Reply:
I have set the state before this. so it won’t be empty

Reply:
got it. btw, you can iterate `result.data.translations` instead of the state.